### PR TITLE
Add 'Logger.log_struct'

### PR DESCRIPTION
**Pull request by tseaver**
Uses #1567 as a base.

### CODE REVIEW

File: gcloud/logging/logger.py [hunk 1]
```diff
@@ -41,3 +41,76 @@ def client(self):
...
+        :type client: :class:`gcloud.logging.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current logger.
+        """
+        client = self._require_client(client)
+
+        data = {
+            'entries': [{
+                'logName': self.full_name,
+                'jsonPayload': info,
```
Comment (dhermes) [inline gcloud/logging/logger.py]: Does `info` need to be serialized first? e.g. w/ slight difference

```python
>>> import json
>>> json.dumps({'key': {}})
'{"key": {}}'
>>> json.dumps({'key': json.dumps('{}')})
'{"key": "\\"{}\\""}'
```

Comment (tseaver) [inline gcloud/logging/logger.py]: Empirically no:  the code as-is works against the live `entries.write` API.

Comment (dhermes) [inline gcloud/logging/logger.py]: OK SGTM

**Comment by tseaver** [comment]
@dhermes this one should be good to go.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**